pkg/scheduler/internal/cache/cache.go

```diff
@@ -222,7 +226,7 @@ func (cache *schedulerCache) UpdateSnapshot(nodeSnapshot *Snapshot) error {
 			// all the nodes are updated before the existing snapshot. We are done.
 			break
 		}
-		if balancedVolumesEnabled && node.info.TransientInfo != nil {
+		if cache.balancedVolumesEnabled && node.info.TransientInfo != nil {
```

Comment:
It looks like we could completely cut off the dependency on the gate with the scheduling framework. Could you look into that instead?
Open a separate PR for it.

https://github.com/kubernetes/kubernetes/blob/271476446b32bfc69a95ea20d8b02d5303128f5e/pkg/scheduler/framework/types.go#L391-L394

Reply:
I undone the change touching the feature gate to leave it for another PR removing dependency on k8s.io/kubernetes/pkg/features